plugins/colordialog/dialogs/colordialog.js

```diff
@@ -276,6 +276,22 @@ CKEDITOR.dialog.add( 'colordialog', function( editor ) {
 		title: lang.title,
 		minWidth: 360,
 		minHeight: 220,
+		onShow: function( evt ) {
+			if ( evt.data.colordialog && evt.data.colordialog.selectionColor ) {
+				var selectionColor = evt.data.colordialog.selectionColor;
+				var colors = evt.sender.parts.contents.getElementsByTag( 'td' );
+
+				dialog.getContentElement( 'picker', 'selectedColor' ).setValue( selectionColor );
+
+				for ( var i = 0; i < colors.count(); i++ ) {
```

Comment:
You can use [`toArray`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_nodeList.html#method-toArray) on `colors` and then use `CKEDITOR.tools.array.forEach` here, for iteration.